pkg/cloud/vsphere/services/kubeconfig/kubeconfig_test.go

```diff
@@ -18,43 +18,77 @@ package kubeconfig_test
 
 import (
 	"encoding/base64"
+	"fmt"
 	"testing"
 
+	"gopkg.in/yaml.v2"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/apis/vsphere/v1alpha1"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/services/kubeconfig"
 )
 
+type Cluster struct {
+	Server string `yaml:"server"`
+}
+type ClusterItem struct {
+	Name    string  `yaml:"name"`
+	Cluster Cluster `yaml:"cluster"`
+}
+
+type Kubeconfig struct {
+	Clusters []ClusterItem `yaml:"clusters"`
+}
+
 func TestNew(t *testing.T) {
 	testcases := []struct {
 		name                 string
 		controlPlaneEndpoint string
+		expected             string
 	}{
 		{
 			name:                 "host:port",
 			controlPlaneEndpoint: "192.168.0.2:6443",
+			expected:             "192.168.0.2:6443",
+		},
+		{
+			name:                 "host:port/path",
+			controlPlaneEndpoint: "192.168.0.2:6443/proxy",
+			expected:             "192.168.0.2:6443",
 		},
 		{
 			name:                 "https://host:port",
 			controlPlaneEndpoint: "https://192.168.0.2:6443",
+			expected:             "https://192.168.0.2:6443",
 		},
 		{
 			name:                 "https://host",
 			controlPlaneEndpoint: "https://192.168.0.2",
+			expected:             "https://192.168.0.2",
 		},
 		{
 			name:                 "https://host:port/path",
 			controlPlaneEndpoint: "https://192.168.0.2:6443/proxy",
+			expected:             "https://192.168.0.2:6443",
 		},
 	}
 
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
-			yaml, err := kubeconfig.New("kubernetes", tc.controlPlaneEndpoint, caKeyPair)
+			y, err := kubeconfig.New("kubernetes", tc.controlPlaneEndpoint, caKeyPair)
 			if err != nil {
 				t.Fatal(err)
 				return
 			}
-			t.Log(yaml)
+			k := Kubeconfig{}
+			err = yaml.Unmarshal([]byte(y), &k)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if k.Clusters[0].Cluster.Server != tc.expected {
```

Comment:
Not sure if I should check for `len(k.Clusters) > 0` here?

Reply:
Hi @pickledrick,

I appreciate the PR, but I must advise you that a lot of the work you're doing will be part of other CAPI components in v1a2. Would you like to schedule some time to chat to make sure you're not spending time on areas that will be deprecated soon?

Reply:
That said, this bug has been driving me nuts, and I have to say thank you from the bottom of my heart for addressing it!